cinema/views.py

```diff
@@ -1,45 +1,107 @@
+from django.shortcuts import get_object_or_404
+from rest_framework import status, generics, mixins, viewsets
 from rest_framework.decorators import api_view
 from rest_framework.response import Response
-from rest_framework import status
+from rest_framework.views import APIView
 
-from django.shortcuts import get_object_or_404
+from cinema.models import Genre, Actor, CinemaHall, Movie
+from cinema.serializers import MovieSerializer, GenreSerializer, ActorSerializer, CinemaHallSerializer
 
-from cinema.models import Movie
-from cinema.serializers import MovieSerializer
 
+class GenreList(APIView):
+    @staticmethod
```

Comment:
The use of `@staticmethod` for the `get` and `post` methods in `GenreList` is unnecessary. These methods should be instance methods, not static methods, as they rely on the `request` object.